volatility/framework/automagic/pdbscan.py

```diff
@@ -75,54 +76,232 @@ def __call__(self, data: bytes, data_offset: int) -> Generator[Tuple[str, Any, b
-
-    if start is None:
-        start = ctx.layers[layer_name].minimum_address
-    if end is None:
-        end = ctx.layers[layer_name].maximum_address
-
-    for (GUID, age, pdb_name, signature_offset) in ctx.layers[layer_name].scan(ctx,
-                                                                               PdbSignatureScanner(pdb_names),
-                                                                               progress_callback = progress_callback,
-                                                                               sections = [(start, end - start)]):
-        mz_offset = None
-        sig_pfn = signature_offset // page_size
-
-        for i in range(sig_pfn, min_pfn, -1):
-            if not ctx.layers[layer_name].is_valid(i * page_size, 2):
+        Args:
+            context: The context on which to operate
+            layer_name: The name of the (contiguous) layer within the context that contains the MZ file
+            offset: The offset in the layer at which the MZ file begins
+            symbol_table_class: The type of symbol class to construct the SymbolTable
+            config_path: New path for the produced symbol table configuration with the config tree
+            progress_callback: Callable called to update ongoing progress
+
+        Returns:
+            None if no pdb information can be determined, else returned the name of the loaded symbols for the MZ
+        """
+        guid, age, pdb_name = cls.get_guid_from_mz(context, layer_name, offset)
+        if config_path is None:
+            config_path = interfaces.configuration.path_join('pdbutility', pdb_name.strip('\x00').replace('.', '_'))
+
+        return cls.load_windows_symbol_table(context, guid, age, pdb_name, symbol_table_class, config_path,
+                                             progress_callback)
+
+    @classmethod
+    def load_windows_symbol_table(cls,
+                                  context: interfaces.context.ContextInterface,
+                                  guid: str,
+                                  age: int,
+                                  pdb_name: str,
+                                  symbol_table_class: str,
+                                  config_path: str = 'pdbutility',
+                                  progress_callback: constants.ProgressCallback = None):
+        """Loads (downlading if necessary) a windows symbol table"""
+
+        filter_string = os.path.join(pdb_name.strip('\x00'), guid.upper() + "-" + str(age))
+
+        isf_path = False
+        # Take the first result of search for the intermediate file
+        for value in intermed.IntermediateSymbolTable.file_symbol_url("windows", filter_string):
+            isf_path = value
+            break
+        else:
+            # If none are found, attempt to download the pdb, convert it and try again
+            PDBUtility.download_pdb_isf(context, guid.upper(), age, pdb_name, progress_callback)
```

Comment:
Is there a reason this is called via the absolute name `PDBUtility` rather than the `cls` variable?

Reply:
There is not, good spot!  5;)